mobly/controllers/android_device_lib/callback_handler_v2.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2022 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""The callback handler V2 module for Android Mobly Snippet Lib."""
+
+from mobly.snippet import callback_handler_base
+from mobly.snippet import errors
+
+
+class CallbackHandlerV2(callback_handler_base.CallbackHandlerBase):
+  """The callback handler V2 class for Android Mobly Snippet Lib."""
+
+  def callEventWaitAndGetRpc(self, callback_id, event_name, timeout_sec):
+    """Waits and returns an existing CallbackEvent for the specified identifier.
+
+    This function calls snippet lib's eventWaitAndGet RPC.
+
+    Args:
+      callback_id: str, the callback identifier.
+      event_name: str, the callback name.
+      timeout_sec: float, the number of seconds to wait for the event.
+
+    Returns:
+      The event dictionary.
+
+    Raises:
+      errors.CallbackHandlerTimeoutError: The expected event does not occur
+        within the time limit.
+    """
+    timeout_ms = int(timeout_sec * 1000)
+    try:
+      return self._event_client.eventWaitAndGet(callback_id, event_name,
+                                                timeout_ms)
+    except Exception as e:
+      if 'EventSnippetException: timeout.' in str(e):
```

Comment:
make this string a const?

also if all the server impl will have its own timeout msg, should this be a property for each child handler to override

Reply:
Made it a constant.

I think we may not add it to a property for base class. As Subclasses currently do not have the constructor to override a property. So a constant string seems a good choice to make subclasses keep simple.